src/System.IO.FileSystem/src/System/IO/PathHelpers.Windows.cs

```diff
@@ -2,80 +2,43 @@
-        //  the user being able to use it to move up directories. Here are some examples eg
-        //    Valid: a..b  abc..d
-        //    Invalid: ..ab   ab..  ..   abc..d\abc..
-        //
-        internal static void CheckSearchPattern(string searchPattern)
+        internal static unsafe int GetRootLength(ReadOnlySpan<char> path)
         {
-            for (int index = 0; (index = searchPattern.IndexOf("..", index, StringComparison.Ordinal)) != -1; index += 2)
+            fixed (char* p = &path.DangerousGetPinnableReference())
             {
-                // Terminal ".." or "..\". File and directory names cannot end in "..".
-                if (index + 2 == searchPattern.Length ||
-                    PathInternal.IsDirectorySeparator(searchPattern[index + 2]))
-                {
-                    throw new ArgumentException(SR.Arg_InvalidSearchPattern, nameof(searchPattern));
-                }
+                return (int)GetRootLength(p, (uint)path.Length);
             }
         }
+    }
 
-        internal static string NormalizeSearchPattern(string searchPattern)
-        {
-            Debug.Assert(searchPattern != null);
-
-            // Win32 normalization trims only U+0020.
-            string tempSearchPattern = searchPattern.TrimEnd(PathHelpers.TrimEndChars);
-
-            // Make this corner case more useful, like dir
-            if (tempSearchPattern.Equals("."))
-            {
-                tempSearchPattern = "*";
-            }
-
-            CheckSearchPattern(tempSearchPattern);
-            return tempSearchPattern;
-        }
-
-        internal static string GetFullSearchString(string fullPath, string searchPattern)
+    internal static partial class PathHelpers
+    {
+        internal static bool ShouldReviseDirectoryPathToCurrent(string path)
         {
-            Debug.Assert(fullPath != null);
-            Debug.Assert(searchPattern != null);
-
-            ThrowIfEmptyOrRootedPath(searchPattern);
-            string tempStr = Path.Combine(fullPath, searchPattern);
-
-            // If path ends in a trailing slash (\), append a * or we'll get a "Cannot find the file specified" exception
-            char lastChar = tempStr[tempStr.Length - 1];
-            if (PathInternal.IsDirectorySeparator(lastChar) || lastChar == Path.VolumeSeparatorChar)
-            {
-                tempStr = tempStr + "*";
-            }
-
-            return tempStr;
+            // In situations where this method is invoked, "<DriveLetter>:" should be special-cased
+            // to instead go to the current directory.
+            return path != null && path.Length == 2 && path[1] == ':';
         }
 
         internal static string TrimEndingDirectorySeparator(string path) =>
             EndsInDirectorySeparator(path) ?
                 path.Substring(0, path.Length - 1) :
                 path;
+
+        public static bool IsPathRooted(string path)
```

Comment:
You're not using Path.IsPathRooted because you want to avoid PathInternal.CheckInvalidPathChars(path); ? That's worth a comment.

Reply:
I would name it IsPathRootedNoCheck or something

Reply:
Ok. Commented.